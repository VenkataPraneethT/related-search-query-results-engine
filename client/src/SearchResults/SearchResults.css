.search-results-container{
    display: flex;
    flex-wrap: wrap;
}

.summary-card-container{
    padding: 10px;
    margin: 10px;
    box-shadow: 0 0 0 1px rgba(0, 0, 0, .1), 0 2px 4px 1px rgba(0, 0, 0, .18);
    flex: 1 0 22%;
    text-align: center;
}

.summary-description-container {
    max-height: 3em;
    text-overflow: ellipsis;
    overflow: hidden;
    max-height: 3.6em;
    line-height: 1.2em;
}

.summary-description-container::after {
    content: '...';
}