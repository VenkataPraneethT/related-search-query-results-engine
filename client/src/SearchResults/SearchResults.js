import React, { PureComponent } from 'react';
// import PropTypes from 'prop-types';
import './SearchResults.css';

export default class SearchResults extends PureComponent {
    render() {
        const {selectedSummaries} = this.props;
        return (
            <div className="search-results-container">
                {selectedSummaries.map((summary, id) => 
                    <div className="summary-card-container" key={id}>
                        <h4>{summary.title}</h4>
                        <p className="summary-description-container">{summary.summary}</p>
                        <hr></hr>
                        <p>{summary.author}</p>
                    </div>
                )}
            </div>
        )
    }
};