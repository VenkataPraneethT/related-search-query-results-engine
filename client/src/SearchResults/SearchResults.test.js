import React from 'react';
import { shallow } from 'enzyme';
import SearchResults from './SearchResults';

describe('<SearchResults />', () => {
  let component;

  beforeEach(() => {
    const props = {
        selectedSummaries:[]
    }
    component = shallow(<SearchResults {...props}/>);
  });

  test('It should mount', () => {
    expect(component.length).toMatchSnapshot();
  });
});
