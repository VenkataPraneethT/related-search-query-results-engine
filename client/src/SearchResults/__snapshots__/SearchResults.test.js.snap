// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> It should mount 1`] = `1`;
